### PR TITLE
Updates to Blueprint System Security Plans

### PR DESCRIPTION
**Pull request by SamBH**
Updates to both System Security Plans (SSPs) to align with September 2021 ISM and BLueprint SSP Annexes published December 2021.

### CODE REVIEW

**Review by zedzedzed — COMMENTED** [review]
Comments apply to both cloud-native and hybrid docs.

File: blueprint/security/system-security-plan.md [hunk 1]
```diff
@@ -15,84 +15,85 @@ CloudSystem.
...
+* DTA – Cloud-Native Blueprint – System Security Plan Annex (September 2021)
+* DTA – Cloud-Native Blueprint – Security Risk Management Plan
+* DTA – Cloud-Native Blueprint – Security Standard Operating Procedures
+* DTA – Cloud-Native Blueprint – Incident Response Plan
 
 The suite of documentation produced to support ACSC’s certification of Azure and Office 365 for PROTECTED have also been leveraged in the development of the CloudSystem, and includes the following :
 
 * 2020 Microsoft Azure IRAP - Azure Security Fundamentals and Cloud Services Assessment
 * Office 365 Security Fundamentals and Cloud Services IRAP Assessment Report 2020
+* 2019 Microsoft Azure IRAP Assessment Report
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:89]: The 2019 reports are getting rather old so agree with removing them.

File: blueprint/security/system-security-plan.md [hunk 2]
```diff
@@ -446,13 +441,19 @@ The CloudSystem provides a role-based access control implementation and associat
 ### Security controls provided by the CloudSystem
 
 * The CloudSystem provides a framework for identity and access management for Azure and Office 365 resources.
+* The CloudSystem includes automation to disable inactive accounts after 45 days.
+* The CloudSystem leverages Azure AD Privileged Identity Management (PIM) to provide Just-in-time administration.
+* Changes to privileged accounts and groups are logged in the Azure AD Audit Log.
+* Intune configures an AppLocker blacklist to prevent administrators from launching web browsers and email clients.
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:447]: Replacement made.

File: blueprint/security/system-security-plan.md [hunk 3]
```diff
@@ -548,29 +549,31 @@ This section is applicable as the CloudSystem includes mobile devices.
...
 
 Bluetooth is enabled as it allows users to pair devices they may required to perform their duties (e.g. conference calls or online meetings).
 
 Users can reset certain security settings in Personal Hotspot and Passcode for situations where the passcode/password may have been compromised.
 
-To provide ease of use and maintain productivity users can download apps from the App Store. The agency can track, and monitor apps found on managed devices that have been enrolled in Intune. Details on how to view and report on discovered apps can be found in the ‘DTA - Cloud-Native Blueprint - Security Standard Operating Procedures (August 2020)’.
+The CloudSystem does not use a VPN on mobile devices, and therefore a direct connection to the internet is used. The Agency may choose to implement a VPN for mobile devices if available from their existing SIG.
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:560]: Updated the wording and included the suggested link.

File: blueprint/security/system-security-plan.md [hunk 4]
```diff
@@ -548,29 +549,31 @@ This section is applicable as the CloudSystem includes mobile devices.
...
+
+To provide ease of use and maintain productivity users can download apps from the App Store. The Agency can track and monitor apps found on managed devices that have been enrolled in Intune. Details on how to view and report on discovered apps can be found in the ‘DTA - Cloud-Native Blueprint - Security Standard Operating Procedures’.
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:562]: Correct - good catch. Have updated wording to:

_Applications are installed from the App Store using the Volume Purchasing Program (VPP) tokens through Apple Business Manager Enrollment. This provides application control for iOS mobile devices, and removes the need for users having to install applications from the App Store through an Apple ID._

File: blueprint/security/system-security-plan.md [hunk 5]
```diff
@@ -548,29 +549,31 @@ This section is applicable as the CloudSystem includes mobile devices.
...
 
-iOS devices are lightly managed and partially comply with the ACSC ‘[Security Configuration Guide - Apple iOS 12 Devices](https://www.cyber.gov.au/publications/security-configuration-guide-apple-ios-12-devices)’ and Information Security Manual (ISM) to maximise usability for the target users. Note: Agencies should do a risk assessment before deciding to change settings relating to mobile devices.
+The CloudSystem does not include the use of privately-owned mobile devices. Only Agency-owned devices are enrolled and allowed to access data.
+
+iOS devices are lightly managed and partially comply with the ACSC ‘[Security Configuration Guide - Apple iOS 14 Devices](https://www.cyber.gov.au/acsc/view-all-content/publications/security-configuration-guide-apple-ios-14-devices)’ and Information Security Manual (ISM) to maximise usability for the target users. Note: Agencies should do a risk assessment before deciding to change settings relating to mobile devices.
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:554]: Agreed - have reworded this section.

File: blueprint/security/system-security-plan.md [hunk 6]
```diff
@@ -548,29 +549,31 @@ This section is applicable as the CloudSystem includes mobile devices.
...
+The CloudSystem does not use a VPN on mobile devices, and therefore a direct connection to the internet is used. The Agency may choose to implement a VPN for mobile devices if available from their existing SIG.
+
+To provide ease of use and maintain productivity users can download apps from the App Store. The Agency can track and monitor apps found on managed devices that have been enrolled in Intune. Details on how to view and report on discovered apps can be found in the ‘DTA - Cloud-Native Blueprint - Security Standard Operating Procedures’.
 
-The risk of lightly managed iOS devices is addressed in the ‘[DTA - Cloud-Native Blueprint - Security Risk Management Plan (August 2020)](security-risk-management-plan.html\)’ at [R17](security-risk-management-plan.html#r17-mobile-device-compromised).
+The risk of lightly managed iOS devices is addressed in the ‘[DTA - Cloud-Native Blueprint - Security Risk Management Plan](security-risk-management-plan.html\)’ at [R17](security-risk-management-plan.html#r17-mobile-device-compromised).
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:564]: Have reworded (and SRMP update is next on the list).

File: blueprint/security/system-security-plan.md [hunk 7]
```diff
@@ -548,29 +549,31 @@ This section is applicable as the CloudSystem includes mobile devices.
...
 
 #### Security controls provided by the CloudSystem
 
 * Microsoft Intune provides MDM and MAM capability.
-* The CloudSystem does not include the use of privately-owned mobile devices. Only Agency-owned devices are enrolled and allowed to access data.
 * The CloudSystem provides Windows 10 for laptops which is hardened in accordance with ACSC guidance. The CloudSystem also provides MDM for iOS but does not fully implement ACSC's guidance for PROTECTED.
 * Microsoft BitLocker provides full disk encryption of CloudSystem mobile devices, implementing Advanced Encryption Standard (AES)-256. Additionally, iOS devices implement AES-256 encryption by default.
 * All information transmitted to and from mobile devices and Office 365 is encrypted.
-* Bluetooth is disabled on Windows 10 devices. Bluetooth is not managed for iOS devices.
+* Bluetooth device type whitelisting is configured on Windows 10 devices. Bluetooth is not managed for iOS devices.
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:572]: Done.

File: blueprint/security/system-security-plan.md [hunk 8]
```diff
@@ -604,13 +607,13 @@ The CloudSystem includes Windows 10 which has been evaluated and therefore the c
 
 ### CloudSystem compliance approach
 
-A Protection Profile (PP) evaluation has been performed on Windows 10 and Microsoft publish deployment and administration guides for each [evaluated operating system](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-platform-common-criteria). The CloudSystem implements the recommendations for the latest evaluated version of Windows 10 (May 2019 Update). This includes sourcing installation media directly from Microsoft and implementing configuration hardening.
+A Protection Profile (PP) evaluation has been performed on Windows 10 and Microsoft publish deployment and administration guides for each [evaluated operating system](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-platform-common-criteria). The CloudSystem implements the recommendations for the latest evaluated version of Windows 10 (November 2019 Update). This includes sourcing installation media directly from Microsoft and implementing configuration hardening.
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:610]: Yes - the November 2019 (1909) update is the latest version listed.

File: blueprint/security/system-security-plan.md [hunk 9]
```diff
@@ -999,12 +1020,14 @@ Local event logs on Windows 10 devices will be lost when endpoints are rebuilt a
   * user or group management
   * use of special privileges.
 * Logs include the date and time of the event, the relevant user or process, the event description, and the ICT equipment involved are recorded.
-* Logs stored in Log Analytics are protected from unauthorised access, modification and deletion by the Azure AD RBAC model. Standard Windows 10 users don’t have access to modify the local event logs.
+* Logs stored in Log Analytics are protected from unauthorised access, modification and deletion by the Azure AD RBAC model. Standard Windows 10 users to do have access to modify the local event logs.
```
[the remark this answers is not in the document]
Comment (SamBH) [reply blueprint/security/system-security-plan.md:1023]: Should have been "do not" - fixed.

File: blueprint/security/hybrid-system-security-plan.md [hunk 10]
```diff
@@ -821,20 +841,28 @@ The HybridSystem leverages Azure AD for controlling system access. All technical
...
+The HybridSystem provides security controls and an identity management framework that can be utilised to manage system access for systems deployed within the HybridSystem. The HybridSystem enforces multi-factor authentication through Conditional Access policies, creates recovery (emergency access) accounts for maintaining access to resources and enforces password policies for accounts created directly in Azure AD. The HybridSystem uses a soft-token - the Microsoft Authenticator app - to reduce the need for purchase, distribution and management of hard-tokens.
+
+The Microsoft Authenticator app cannot be centrally managed to require the enforcement of a PIN or biometric unlock, and is not considered verifier impersonation resistant.
+
+The Protected User security group is not available in Azure AD and is therefore not used by the HybridSystem.
```
Comment (oobedan) [inline blueprint/security/hybrid-system-security-plan.md:848]: suggest adding that it is an agency responsibility to configure the protected users group for on-prem AD instance that is sync'd to Azure AD, as this reads that it doesn't exist for hybird.

**Comment by SamBH** [comment]
Responded to all comments from Michael and both documents were updated accordingly.

File: blueprint/security/hybrid-system-security-plan.md [hunk 11]
```diff
@@ -821,20 +841,28 @@ The HybridSystem leverages Azure AD for controlling system access. All technical
...
 * Azure AD is configured to require all users to be authenticated before granting access.
-* Azure MFA is enforced for all standard and privileged users.
-* MFA requires complex password and One Time Password (OTP) from Microsoft Authenticator App (soft token).
-* None of the authentication factors on their own can be used for single-factor authentication to another system.
-* Azure MFA is enforced for all users accessing Office 365 content.
-* Azure AD self-service password reset requires users to verify their identity before resetting their password in accordance with password complexity requirements.
-* Local Area Network (LAN) Manager is not used by the HybridSystem, however New Technology LAN Manager (NTLM) is used within the system.
-* Credentials are stored within Azure AD. Azure AD Identity Protection is enabled to detected leaked passwords.
+* Azure MFA is enforced for all standard and privileged users accessing M365 services.
+* MFA requires Azure AD password and Microsoft Authenticator app (either acceptance of push notification or entry of OTP.
```
Comment (oobedan) [inline blueprint/security/hybrid-system-security-plan.md:856]: missing a closing )

File: blueprint/security/hybrid-system-security-plan.md [hunk 12]
```diff
@@ -821,20 +841,28 @@ The HybridSystem leverages Azure AD for controlling system access. All technical
...
+* Azure AD password complexity enforces a minimum character length of 8 characters.
+* All authentication attempts are logged in Azure AD Sign-ins.
+* Azure AD logs are forwarded to a Log Analytics workspace for long-term secure retention.
+* Legacy authentication methods are disabled in the Windows 10 SOE following the ACSC hardening guide.
+* Credentials are stored within Azure AD and the on-premises AD.
+* Standard Windows & iOS functionality is to obscure passwords during logon.
+* Windows Defender Credential Guard is enabled for the Windows 10 SOE.
```
Comment (oobedan) [inline blueprint/security/hybrid-system-security-plan.md:863]: should it be mentioned if the device supports it or is this assumed?

File: blueprint/security/hybrid-system-security-plan.md [hunk 13]
```diff
@@ -916,31 +944,28 @@ Local sites and data stores | On-premises SharePoint Server
...
 * Windows Update verifies the integrity of patches before installing them.
+* Microsoft Defender for Endpoint provides a continuous vulnerability management capability for all Windows 10 devices.
 
 #### Residual controls to be addressed by the Agency
 
-* The Agency is responsible for system patching for all systems deployed within the HybridSystem.
-* The Agency is responsible for any firmware patching dependent on the specific hardware model chosen by them.
-* The Agency is responsible for viewing the reporting and alerts and rectification of faults as they occur.
+* The Agency is responsible for maintaining and auditing a software register.
+* The Agency is responsible for patching any third-party applications  deployed to Windows 10 devices.
```
Comment (oobedan) [inline blueprint/security/hybrid-system-security-plan.md:966]: seems to be a double space in beteen 'applications' and 'deployed'

File: blueprint/security/hybrid-system-security-plan.md (hunk 10 above)
Comment (SamBH) [reply blueprint/security/hybrid-system-security-plan.md:846]: Agreed. Updated compliance approach and added to residual controls for Agency.

File: blueprint/security/hybrid-system-security-plan.md (hunk 11 above)
Comment (SamBH) [reply blueprint/security/hybrid-system-security-plan.md:856]: Fixed for both hybrid and cloud-only SSPs.

File: blueprint/security/hybrid-system-security-plan.md (hunk 12 above)
Comment (SamBH) [reply blueprint/security/hybrid-system-security-plan.md:863]: The hardware requirements in the Client Devices design specify what is required to support Credential Guard so I think we can take it as assumed here.

File: blueprint/security/hybrid-system-security-plan.md (hunk 13 above)
Comment (SamBH) [reply blueprint/security/hybrid-system-security-plan.md:966]: Fixed - along with some others I found in both SSPs.

File: blueprint/security/system-security-plan.md [hunk 14]
```diff
@@ -372,14 +370,10 @@ Development and maintenance of security documentation is applicable to the Cloud
...
-Cloud-Only provides an SSP (this document), SSP Annex (formerly the Statement of Applicability (SoA)), SRMP, Incident Response Plan (IRP), Standard Operating Procedures (SOPs) and other operational documentation to assist in the understanding of the Cloud-Only system and the security controls included.
-
 #### Security controls provided by the CloudSystem
 
-* The CloudSystem provides SOPs for administrators and support staff to understand, monitor and operate the CloudSystem-provided security controls.
-* The CloudSystem includes detailed design, configuration, operational and support documentation.
-* The CloudSystem provides security documentation for input into an Agency’s security processes.
-* The CloudSystem's security documentation and notification of subsequent changes is communicated by DTA to Agencies who have implemented the CloudSystem.
+* All security documentation produced by DTA for the CloudSystem has been updated within the last year and include the 'last updated' date.
+* DTA releases updated documentation to their stakeholders via the Protected Utility Community website.
```
Comment (oobedan) [inline blueprint/security/system-security-plan.md:374]: Is this still the case @zedzedzed

Comment (zedzedzed) [reply blueprint/security/system-security-plan.md:374]: Nope, it should be changed to:

```* DTA provide updates to documentation via the Protected Utility website at [desktop.gov.au](https://desktop.gov.au/). Release logs and change sets between versions can be obtained at [Github](https://github.com/govau/desktop.gov.au).```

Comment (SamBH) [reply blueprint/security/system-security-plan.md:374]: I'll update that now.

**Comment by SamBH** [comment]
@zedzedzed All feedback from yourself and @oobedan has been actioned. If there is nothing more I believe this PR is ready for approval.

**Review by zedzedzed — APPROVED** [review]

**Pull request merged**